packages/@lwc/engine-core/src/framework/modules/attr-unless-prop.ts

```diff
@@ -43,7 +43,7 @@ export function patchAttrUnlessProp(
         if (old !== cur) {
             const propName = htmlAttributeToProperty(name);
             if (propName in elm!) {
-                (elm as any)[propName] = cur;
+                setProperty(elm, name, cur);
```

Comment:
@caridy  @nolanlawson engine-dom`'s `setProperty()` does the same thing under the hood; however, it does do some extra things for `engine-server`. Was that the concern being addressed with this suggested change?

Reply:
Locker can intercept any renderer API; see #2861. 🙂